src/lib.rs

```diff
@@ -259,5 +261,42 @@ pub fn getrandom(dest: &mut [u8]) -> Result<(), Error> {
     if dest.is_empty() {
         return Ok(());
     }
-    imp::getrandom_inner(dest)
+    imp::getrandom_inner(
+        // SAFETY: `MaybeUninit` have same size and alignment as wrapped value
+        // and it is safe to convert init value to MaybeUninit.
+        // See https://doc.rust-lang.org/stable/std/mem/union.MaybeUninit.html#layout
+        unsafe {
+            core::slice::from_raw_parts_mut(dest.as_mut_ptr() as *mut MaybeUninit<u8>, dest.len())
+        },
+    )
+}
+
+/// Fill `dest` with random bytes from the system's preferred random number
+/// source.
+///
+/// This function returns an error on any failure, including partial reads.
+/// In case of error, underlying values should be treated as uninit.
+/// If `dest` is empty, `getrandom` immediately returns success, making no calls to the
+/// underlying operating system.
+/// If call is successful, values in dest is initialized.
+///
+/// Blocking is possible, at least during early boot; see module documentation.
+///
+/// In general, `getrandom_to_uninit` will be fast enough for interactive usage, though
+/// significantly slower than a user-space CSPRNG; for the latter consider
+/// [`rand::thread_rng`](https://docs.rs/rand/*/rand/fn.thread_rng.html).
+pub fn getrandom_to_uninit(dest: &mut [MaybeUninit<u8>]) -> Result<(), Error> {
+    if dest.is_empty() {
+        return Ok(());
+    }
+    let res = imp::getrandom_inner(dest);
+
+    // Help miri to catch mistakes.
+    // If user calls `assume_init()` on elements,
+    // miri would show error.
+    #[cfg(miri)]
+    if res.is_err() {
+        dest.fill(MaybeUninit::uninit());
+    }
```

Comment:
I am thinking that maybe this is better to be put into `rand` crate and let `rand` use `getrandom_to_uninit` even for initialized slices.